tests/components/kem/test_config_flow.py

```diff
@@ -0,0 +1,194 @@
+"""Test the Kem config flow."""
+
+from unittest.mock import AsyncMock, patch
+
+from aiokem import AuthenticationCredentialsError
+
+from homeassistant import config_entries
+from homeassistant.components.kem.const import DOMAIN
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import Mock, MockConfigEntry
+
+
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with (
+        patch("homeassistant.components.kem.data.AioKem.authenticate"),
+        patch(
+            "homeassistant.components.kem.async_setup_entry",
+            return_value=True,
+        ) as mock_setup_entry,
+    ):
+        mock_setup_entry.runtime_data = Mock()
+        mock_setup_entry.runtime_data.kem = AsyncMock()
+        result2 = await hass.config_entries.flow.async_configure(
```

Comment:
I would recommend to overwrite `result` as you never need the old one again and avoid mistakes